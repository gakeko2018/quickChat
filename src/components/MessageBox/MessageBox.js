import React from "react";
import { StyleSheet, Text, View, Image } from "react-native";
import { colors, margin, padding } from "../../styles/base";
import PropTypes from "prop-types";

const MessageBox = ({ username, message }) => {
  const {
    id,
    text,
    user: { avatarUrl, name }
  } = message;
  return (
    <View style={username === name ? styles.userChatBox : styles.chatBox}>
      <View style={styles.avatarBox}>
        <Image source={{ uri: avatarUrl }} style={styles.avatarImage} />
      </View>
      <Text
        style={
          username === name
            ? styles.userMessageBodyText
            : styles.messageBodyText
        }
      >
        {text}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  userChatBox: {
    minWidth: "100%",
    flexDirection: "row-reverse",
    marginBottom: margin.md
  },
  chatBox: {
    minWidth: "100%",
    flexDirection: "row",
    marginBottom: margin.md
  },
  avatarBox: {
    height: 40,
    width: 40
  },
  avatarImage: {
    position: "absolute",
    top: 0,
    left: 0,
    bottom: 0,
    right: 0,
    borderRadius: 20
  },
  messageBodyText: {
    backgroundColor: colors.tertiaryBackground,
    borderRadius: 10,
    margin: margin.sm,
    padding: padding.sm,
    maxWidth: "60%",
    overflow: "hidden",
    color: colors.primaryFontColor
  },
  userMessageBodyText: {
    backgroundColor: colors.secondaryFontColor,
    color: colors.primaryFontColor,
    borderRadius: 10,
    margin: margin.sm,
    padding: padding.sm,
    maxWidth: "60%",
    overflow: "hidden"
  }
});

MessageBox.propTypes = {
  username: PropTypes.string,
  message: PropTypes.object
};

export default MessageBox;
