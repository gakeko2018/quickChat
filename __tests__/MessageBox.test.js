import React from "react";
import { shallow } from "enzyme";
import MessageBox from "../src/components/MessageBox/MessageBox";

import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

let wrapper;
const username = "John Doe";
const item = {
  id: 4,
  text: "omw",
  timestamp: 1554227223,
  user: {
    avatarUrl:
      "https://images-na.ssl-images-amazon.com/images/M/MV5BNTczMzk1MjU1MV5BMl5BanBnXkFtZTcwNDk2MzAyMg@@._V1_UY256_CR2,0,172,256_AL_.jpg",
    id: 2,
    name: "The Fresh Prince"
  }
};

beforeEach(() => {
  wrapper = shallow(<MessageBox username={username} message={item} />);
});

describe("MessageBox rendering", () => {
  it(" MessageBox renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
