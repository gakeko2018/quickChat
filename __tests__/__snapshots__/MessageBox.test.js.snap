// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBox rendering  MessageBox renders correctly 1`] = `ShallowWrapper {}`;
